prefect_kubernetes/worker.py

```diff
@@ -0,0 +1,671 @@
+    import kubernetes
+    import kubernetes.client
+    import kubernetes.client.exceptions
+    import kubernetes.config
+    from kubernetes.client import ApiClient, BatchV1Api, CoreV1Api, V1Job, V1Pod
+    from prefect.client.schemas import FlowRun
+
+else:
+    kubernetes = lazy_import("kubernetes")
+
+
+def _get_default_job_manifest_template() -> Dict[str, Any]:
+    """Returns the default job manifest template used by the Kubernetes worker."""
+    return {
+        "apiVersion": "batch/v1",
+        "kind": "Job",
+        "metadata": {
+            "labels": "{{ labels }}",
+            "namespace": "{{ namespace }}",
+            "generateName": "{{ name }}-",
+        },
+        "spec": {
+            "ttlSecondsAfterFinished": "{{ finished_job_ttl }}",
+            "template": {
+                "spec": {
+                    "parallelism": 1,
+                    "completions": 1,
+                    "restartPolicy": "Never",
+                    "serviceAccountName": "{{ service_account_name }}",
+                    "containers": [
+                        {
+                            "name": "prefect-job",
+                            "env": "{{ env }}",
+                            "image": "{{ image }}",
+                            "imagePullPolicy": "{{ image_pull_policy }}",
+                            "args": "{{ command }}",
+                        }
+                    ],
+                }
+            },
+        },
+    }
+
+
+def _get_base_job_manifest():
+    """Returns a base job manifest to use for manifest validation."""
+    return {
+        "apiVersion": "batch/v1",
+        "kind": "Job",
+        "metadata": {"labels": {}},
+        "spec": {
+            "template": {
+                "spec": {
+                    "parallelism": 1,
+                    "completions": 1,
+                    "restartPolicy": "Never",
+                    "containers": [
+                        {
+                            "name": "prefect-job",
+                        }
+                    ],
+                }
+            }
+        },
+    }
+
+
+class KubernetesImagePullPolicy(enum.Enum):
+    """Enum representing the image pull policy options for a Kubernetes job."""
+
+    IF_NOT_PRESENT = "IfNotPresent"
+    ALWAYS = "Always"
+    NEVER = "Never"
+
+
+class KubernetesWorkerJobConfiguration(BaseJobConfiguration):
+    """
+    Configuration class used by the Kubernetes worker.
+
+    An instance of this class is passed to the Kubernetes worker's `run` method
+    for each flow run. It contains all of the information necessary to execute
+    the flow run as a Kubernetes job.
+
+    Attributes:
+        name: The name to give to created Kubernetes job.
+        command: The command executed in created Kubernetes jobs to kick off
+            flow run execution.
+        env: The environment variables to set in created Kubernetes jobs.
+        labels: The labels to set on created Kubernetes jobs.
```

Comment:
suggestion: consider allowing users to configure `annotations` as well, as they are sometimes used to provide settings for other controllers. we can also omit that for now and see if anyone asks for it, since adding new parameters is a backward-compatible change